src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+    }
+  },
+
+  components: {
+    SyncLoader
+  },
+
+  data: function () {
+    return {
+      warning: true,
+      /**
+       * Loading indicator data. Used to tell the component whether it should display the loading status or not,
+       * which color to use, etc.
+       */
+      loading: true,
+      vueSpinner: {
+        color: '#5e9aff',
+        size: '.2em'
+      },
+      /**
+       * The Cytoscape layout options. By default it is set to Dagre. When switching the layout in the UI, it will
+       * replace this by a hierarchicalOptions or another options object.
+       */
+      layoutOptions: dagreOptions,
+      /**
+       * Available layout engines. These are used by components in the UI to allow the user to choose which layout
+       * to use in the UI
+       */
+      layoutEngines: [
+        'dagre',
+        'hierarchical'
+      ],
+      /**
+       * Cytoscape layout data. Controls things like the current layout, whether layout is frozen or not, etc.
+       */
+      layoutName: 'dagre',
+      layoutStopped: true,
+      layoutReady: false,
+      freeze: false,
+      /**
+       * Cytoscape tooltip plugin object.
+       */
+      tippy: null,
+      /**
+       * Graph data. These values are empty by default, then are populated later via watchers.
+       */
+      graphData: {
+        nodes: [],
+        edges: []
+      },
+      cytoscapeInstance: null
+    }
+  },
+
+  /**
+   * The watchers created by this component.
+   *
+   * The workflows property is mapped by Vuex (mapState). Whenever Vuex changes it, due to
+   * changes to the store, we are automatically notified. Even if the workflows object reference
+   * was not changed, but one of its properties was (deep: true).
+   *
+   * When the workflowUpdated method is executed, it will calculate the graphData, and change it
+   * once done.
+   *
+   * The graphData watcher will react to changes (not deep!) of graphData, and call the function
+   * updateGraph (debounced). This function is responsible for then - as per its name - update
+   * the graph, the layout, any other setting necessary once the nodes and edges have been added
+   * or removed.
+   */
+  watch: {
+    workflows: {
+      handler: function (newValue) {
+        this.workflowUpdated(newValue)
+      },
+      deep: true
+    },
+    graphData: function () {
+      this.updateGraph()
+    }
+  },
+
+  /**
+   * When this component is created, we simply create debounced functions
+   */
+  created () {
+    this.updateGraph = debounce(this.updateGraph_, 100)
+    this.workflowUpdated = debounce(this.workflowUpdated_, 100)
+  },
+
+  mounted () {
+    // create a div for cytoscape, this is an approach similar to vue-cytoscape
+    // but the vue-cytoscape component offers more than what we need, so we imitate
+    // part of its behaviour here, and ignore the rest and handle the graph and cytoscape
+    // settings ourselves.
+    const element = document.createElement('div')
```

Comment:
This is the approach of `vue-cytoscape`. We didn't need the whole library, so removed it from `package.json`, and added a few lines here to set up the `<div>` for Cytoscape.